src/viam/app/data/client.py

```diff
@@ -65,36 +77,103 @@ async def delete_binary_data_by_filter(self, filter: Optional[Filter]) -> None:
+        """Remove tags from binary data using a filter
+
+        Args:
+            tags (List[str]): List of tags to remove from specified binary data
+            filter (viam.app.proto.Filter): Filter specifying binary data to untag. If no filter is provided, all data will be tagged
+
+        Raises:
+            GRPCError: if no tags are provided
+        """
+        filter = filter if filter else Filter()
+        request = RemoveTagsFromBinaryDataByFilterRequest(filter=filter, tags=tags)
+        _: RemoveTagsFromBinaryDataByFilterResponse = await self._data_client.RemoveTagsFromBinaryDataByFilter(
+            request,
+            metadata=self._metadata
+        )
+
+    async def tags_by_filter(self, filter: Optional[Filter] = None) -> List[str]:
+        """Get a list of tags using a filter
+
+        Args:
+            filter (viam.app.proto.Filter): Filter specifying data to retreive from. If no filter is provided, all data tags will return
+
+        Returns:
+            List[str]: The list of tags
+        """
+        filter = filter if filter else Filter()
+        request = TagsByFilterRequest(filter=filter)
+        response: TagsByFilterResponse = await self._data_client.TagsByFilter(request, metadata=self._metadata)
+        return response.tags
+
+    # later
     async def add_bounding_box_to_image_by_id(self):
         raise NotImplementedError()
 
-    # to be defined and implemented last
+    # later
     async def remove_bounding_box_from_image_by_id(self):
         raise NotImplementedError()
 
-    async def bounding_box_labels_by_filter(self, filter: Optional[Filter]) -> List[str]:
-        raise NotImplementedError()
+    async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -> List[str]:
+        """Get a list of bounding box labels using a filter
 
-    # TODO (RSDK-3637): confirm arguments
-    async def data_capture_upload(self, metadata: Optional[UploadMetadata], sensor_contents: Optional[List[SensorData]]) -> None:
-        raise NotImplementedError()
+        Args:
+            filter (viam.app.proto.Filter): Filter specifying data to retreive from. If no filter is provided, all labels will return
 
-    # TODO (RSDK-3637): confirm arguments
-    async def file_upload(self, metadata: Optional[UploadMetadata], file_contents: Optional[FileData]) -> None:
-        raise NotImplementedError()
+        Returns:
+            List[str]: The list of bounding box labels
+        """
+        filter = filter if filter else Filter()
+        request = BoundingBoxLabelsByFilterRequest(filter=filter)
+        response: BoundingBoxLabelsByFilterResponse = await self._data_client.BoundingBoxLabelsByFilter(request, metadata=self._metadata)
+        return response.labels
+
+    async def data_capture_upload(self, metadata: UploadMetadata , sensor_contents: Optional[List[SensorData]] = None) -> None:
```

Comment:
Awesome to see that you went ahead and implemented these two! We should add docstring comments explaining the semantics of these a bit. Also, have you tested them at all? Probably good to do so.

It's also totally fine to leave these for a subsequent PR, and in fact maybe good to do so if you think that testing/documenting might take some time.

Reply:
As per my Slack message: I meant to leave this out of this PR! The functions work and upload data but trying to decipher the required parameters is its own mess that is probably better solved in the data_sync.proto ticket. I think I'm going to revert these back to raising unimplemented errors and commit again. I have a good local testing suite set up for these though that highlights what we might want to include